pkg/machinery/config/types/v1alpha1/v1alpha1_validation.go

```diff
@@ -204,6 +199,39 @@ func (c *ClusterConfig) Validate() error {
 	return result.ErrorOrNil()
 }
 
+// ValidateCNI validates CNI config.
+func ValidateCNI(cni config.CNI) error {
+	var result *multierror.Error
+
+	switch cni.Name() {
+	case constants.FlannelCNI:
+		fallthrough
+	case constants.NoneCNI:
+		if len(cni.URLs()) != 0 {
+			err := fmt.Errorf(`"urls" field should be empty for %q CNI`, cni.Name())
+			result = multierror.Append(result, err)
+		}
+
+	case constants.CustomCNI:
+		if len(cni.URLs()) == 0 {
+			err := fmt.Errorf(`"urls" field should not be empty for %q CNI`, cni.Name())
```

Comment:
this should be warning for now, probably with a note that we can make this an error in 0.12

the reason is that Talos 0.9 accepted `custom` CNI with empty URL list and some users took advantage of that. we fixed the validation in 0.10, but we shouldn't break backwards compatiblity this way, as after upgrade invalid config would lead to node stuck in reboot loop